### PR TITLE
ARM: Implement `teq` `ALU` instruction

### PR DESCRIPTION
**Pull request by guerinoni**
This don't cover all case because we are not taking care of changing all CPU register and other marginal stuff,
but it's a start :).

Signed-off-by: Federico Guerinoni <guerinoni.federico@gmail.com>

### CODE REVIEW

File: src/arm7tdmi.rs [hunk 1]
```diff
@@ -227,13 +234,29 @@ impl Arm7tdmi {
 
         match ArmModeAluInstruction::from(alu_opcode) {
             ArmModeAluInstruction::Mov => self.mov(rd as usize, op2),
-            _ => todo!(),
+            ArmModeAluInstruction::Teq => {
+                let val = self.teq(rn, op2);
+                self.set_registers(rd, val);
```
Comment (gallottino) [inline src/arm7tdmi.rs:239]: Rn shouldn't be changed. I think this opcode is only about flags.
[here](https://www.problemkaputt.de/gbatek.htm#armopcodesdataprocessingalu) is:
`Void = Rn XOR Op2`

In other instructions there is something like this:
`Rd = Rn OR Op2`

File: src/arm7tdmi.rs [hunk 2]
```diff
@@ -277,4 +300,18 @@ mod tests {
...
+        let op_code: u32 = 0b1110_0001_0010_1001_0011_0000_0000_0000;
+        let mut cpu = Arm7tdmi::new(vec![]);
+
+        let (_, instruction) = cpu.decode(op_code);
+        assert_eq!(instruction, ArmModeInstruction::DataProcessing1);
+
+        cpu.execute(op_code, instruction);
+        let rn = 0b0011;
+        let rd_xor_op2 = 0b1001;
+        assert_eq!(cpu.registers[rn], rd_xor_op2);
```
Comment (gallottino) [inline src/arm7tdmi.rs:315]: For the above reason this test should be different.

**Review by gallottino — CHANGES_REQUESTED** [review]

File: src/arm7tdmi.rs (hunk 1 above)
Comment (guerinoni) [reply src/arm7tdmi.rs:239]: Yes, indeed is not modified

File: src/arm7tdmi.rs (hunk 2 above)
Comment (guerinoni) [reply src/arm7tdmi.rs:315]: The right name is `rd` I'm updating it

File: src/arm7tdmi.rs (hunk 1 above)
Comment (gallottino) [reply src/arm7tdmi.rs:239]: Sorry, I would to say Rd.

Comment (guerinoni) [reply src/arm7tdmi.rs:239]: What this operator should do with the result of that XOR?

Comment (gallottino) [reply src/arm7tdmi.rs:239]: I think  this opcode changes only the flags in CPSR

Comment (guerinoni) [reply src/arm7tdmi.rs:239]: In which way? You mean something like `cpsr = (result of XOR)` (since cpsr is a u32...)

Comment (guerinoni) [reply src/arm7tdmi.rs]: I figured out the right way :)

**Review by gallottino — APPROVED** [review]

**Pull request merged**